bpf-prog/bpf-tc/pkg/ingress/bpf/ingress.c

```diff
@@ -0,0 +1,75 @@
+#include <linux/bpf.h>
+#include <bpf/bpf_helpers.h>
+#include <linux/pkt_cls.h>
+#include <linux/if_ether.h>
+#include <linux/ipv6.h>
+#include <netinet/in.h>
+#include <netinet/tcp.h>
+#include <linux/if_ether.h>
+#include <string.h>
+#include <stdint.h>
+#include <stdbool.h>
+#include "../../../include/helper.h"
+
+SEC("classifier")
+int linklocal_to_gua(struct __sk_buff *skb)
+{
+    // Define the global unicast address 2603:1062:0000:0001:fe80:1234:5678:9abc
+    const struct in6_addr GLOBAL_UNICAST_ADDR = {{{0x26, 0x03, 0x10, 0x62, 0x00, 0x00, 0x00, 0x01, 0xfe, 0x80, 0x12, 0x34, 0x56, 0x78, 0x9a, 0xbc}}};
+    // Define the link-local address fe80::1234:5678:9abc
+    const struct in6_addr LINKLOCAL_ADDR = {{{0xfe, 0x80, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x12, 0x34, 0x56, 0x78, 0x9a, 0xbc}}};
+
+    struct in6_addr src_addr;
+    struct ipv6hdr ipv6_hdr;
+
+    int ret = bpf_skb_load_bytes(skb, ETH_HLEN + offsetof(struct ipv6hdr, saddr), &src_addr, sizeof(src_addr));
+    if (ret != 0)
+    {
+        bpf_printk("bpf_skb_load_bytes failed with error code %d.\n", ret);
+        return TC_ACT_UNSPEC;
+    }
+
+    int ret_hdr = bpf_skb_load_bytes(skb, ETH_HLEN, &ipv6_hdr, sizeof(ipv6_hdr));
+    if (ret_hdr != 0)
+    {
+        bpf_printk("bpf_skb_load_bytes failed with error code %d.\n", ret_hdr);
+        return TC_ACT_UNSPEC;
+    }
+
+    // Check if the packet is TCP
+    if (ipv6_hdr.nexthdr != IPPROTO_TCP)
+        return TC_ACT_UNSPEC;
+
+    // Check the source address to determine if it is Link Local
+    if (compare_ipv6_addr(&src_addr, &LINKLOCAL_ADDR))
+    {
+        bpf_printk("Source address is a link local address. Setting new addr to global unicast.\n");
```

Comment:
skip printing this and enable only for debug.. we can add DEBUG macro and that should be define only if debug mode is enabled. it can be cmdline parameter of go program

Reply:
added DEBUG macro, it can be defined if we pass DEBUG=true as a build-arg when building the image. 

in order to set the macro from the go program cmdline we'd have to use cgo and import C

Reply:
wanted to run with existing binary with debug mode and not another build.. may be we can take as separate PR but it would be useful feature to add to debug in customer cluster when we found ebpf program not working